### DISCUSSION TITLE
Coil cache living alongside Glide cache

### DISCUSSION THREAD

**Discussion by drinkthestars** [coil-kt/coil#1193] [Q&A]
Hi everyone,

We currently use Glide exclusively in our project but are looking to use Coil for features that use Jetpack Compose!

One concern we had was around the Glide image cache potentially living alongside the Coil image cache:
- Would Glide's & Coil's image caches be completely separate from one another, such that loading the _same_ image using both libs would populate both caches?
- We have seen OOM errors in our app due to Glide's image caching at some point. Would introducing Coil increase that risk perhaps?
- Longshot: is there any way to "unify" both libraries' caches?

Any advice/thoughts are much appreciated! 🙏🏼

**Comment by colinrtwhite** [comment]
I was thinking about this for a few days and unfortunately I don't think there's a perfect solution for this.

For disk caching you could potentially use Coil's `DiskCache` to create a custom [disk cache factory](https://bumptech.github.io/glide/doc/configuration.html#disk-cache) for Glide. You could also go the opposite way and use Glide's disk cache with Coil, though it's single-threaded and doesn't support cache headers; Coil's implementation is multithreaded and supports cache headers.

Memory caching is a bit tougher since Coil's and Glide's memory caches are fairly different. Glide removes images from the cache when it returns them whereas Coil's allows sharing resources and doesn't remove them automatically. You could try fitting Coil's `MemoryCache` implementation into Glide's [memory cache factory](https://bumptech.github.io/glide/doc/configuration.html#memory-cache) or vice versa. However, the safest thing might be to allocate 1/2 of the memory typically allocated to image loaders to both Coil and Glide. Coil [defaults to 20%](https://github.com/coil-kt/coil/blob/090a6d7ab2baced576adc3e3905aa07d03b5491b/coil-base/src/main/java/coil/util/Utils.kt#L228) so setting each image loader to 10% should probably be safe. This has downsides, though, as you'll be able to store less images in each individual memory cache and they won't share images.

Hope that helps!

**Comment by drinkthestars** [comment]
Thank you, this is super helpful! Will give some of these a shot! 🙏🏼
